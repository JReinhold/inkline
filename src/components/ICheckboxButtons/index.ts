export { default as ICheckboxButtons } from '@inkline/inkline/components/ICheckboxButtons/ICheckboxButtons.vue';
