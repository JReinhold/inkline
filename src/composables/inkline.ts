import { inject } from 'vue';
import { InklineKey } from '@inkline/inkline';
import type { InklineService } from '@inkline/inkline/plugin';

export function useInkline() {
    return inject(InklineKey) as InklineService;
}
