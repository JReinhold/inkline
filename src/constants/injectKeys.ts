import { InjectionKey } from 'vue/dist/vue';
import type { InklineService } from '@inkline/inkline/plugin';
import type { ToastService } from '@inkline/inkline/plugins';
import type { SvgNode } from '@inkline/inkline/types';

export const InklineKey = Symbol('inkline') as InjectionKey<InklineService>;

export const InklineIconsKey = Symbol('inklineIcons') as InjectionKey<Record<string, SvgNode>>;

export const InklineToastKey = Symbol('inklineToast') as InjectionKey<ToastService>;
