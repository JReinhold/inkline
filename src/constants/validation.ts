import { FormField, ReservedFormKeys, ResolvedFormSchemaState } from '@inkline/inkline/types';

export const defaultValidationStateValues: ResolvedFormSchemaState = {
    pristine: true,
    dirty: false,
    untouched: true,
    touched: false,
    valid: true,
    invalid: false,
    errors: []
};

export const defaultValidationFieldValues: FormField<string> = {
    value: '',
    validators: []
};

export const reservedValidationFields: Array<string | ReservedFormKeys> = [
    'value',
    'validators',
    'pristine',
    'dirty',
    'untouched',
    'touched',
    'valid',
    'invalid',
    'errors'
];
