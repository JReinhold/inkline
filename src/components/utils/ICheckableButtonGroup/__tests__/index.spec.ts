import { render } from '@testing-library/vue';
import { ICheckableButtonGroup } from '@inkline/inkline/components/utils';
import { InklineKey } from '@inkline/inkline/constants';
import { createInkline } from '@inkline/inkline/__tests__/utils';

describe('Components', () => {
    describe('ICheckableButtonGroup', () => {
        const props = {
            color: 'light',
            size: 'md'
        };

        const slots = {
            default: [
                '<i-button>Apple</i-button>',
                '<i-button>Watermelon</i-button>',
                '<i-button>Peach</i-button>',
                '<i-button>Mango</i-button>'
            ]
        };

        it('should be named correctly', () => {
            expect(ICheckableButtonGroup.name).toEqual('ICheckableButtonGroup');
        });

        it('should render correctly', () => {
            const wrapper = render(ICheckableButtonGroup, {
                global: {
                    provide: {
                        [InklineKey as symbol]: createInkline()
                    }
                },
                props,
                slots
            });
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
