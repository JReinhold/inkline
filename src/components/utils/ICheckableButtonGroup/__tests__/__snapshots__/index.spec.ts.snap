// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > ICheckableButtonGroup > should render correctly 1`] = `
"<div class=\\"checkable-button-group -light -md -undefined -default\\">
  <!-- @slot default Slot for default content -->
  <i-button>Apple</i-button>
  <i-button>Watermelon</i-button>
  <i-button>Peach</i-button>
  <i-button>Mango</i-button>
</div>"
`;
